### PR TITLE
Add UTXO changes 2.0 endpoints

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

**Review by qrayven — DISMISSED** [review]

File: sdk/src/client/node_api/core/routes.rs [hunk 1]
```diff
@@ -398,6 +413,18 @@ impl ClientInner {
             .await
     }
 
+    /// Get all UTXO changes of a given slot by its index.
+    /// GET /api/core/v3/commitments/by-index/{index}/utxo-changes
+    pub async fn get_utxo_changes_by_slot_index(&self, slot_index: &SlotIndex) -> Result<UtxoChangesResponse> {
```
Comment (Alex6323) [inline sdk/src/client/node_api/core/routes.rs]: `SlotIndex` is `Copy`, so the API is probably more convenient if it's not taking a `&`?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Definitely, copy pasted mistake

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -374,6 +374,21 @@ impl ClientInner {
             .await
     }
 
+    /// Get all UTXO changes of a given slot by slot commitment ID.
+    /// GET /api/core/v3/commitments/{commitmentId}/utxo-changes
+    pub async fn get_utxo_changes_by_slot_commitment_id(
+        &self,
+        slot_commitment_id: &SlotCommitmentId,
```
Comment (Alex6323) [inline sdk/src/client/node_api/core/routes.rs:381]: why taking a `&`  to a type that is `Copy`? I know what you'll say ...

File: sdk/src/client/node_api/core/routes.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Changed

File: sdk/src/client/node_api/core/routes.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:381]: Still, 40 bytes is not 4. Just because it's Copy doesn't mean you can't use refs anymore.

Comment (Alex6323) [reply sdk/src/client/node_api/core/routes.rs:381]: I know ... just wanted to make sure it's really due to performance considerations.

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**